package/sbp_nmea2000_bridge/src/sbp_nmea2000_bridge.cpp

```diff
@@ -0,0 +1,917 @@
+        N2kMsg.Add4ByteUInt(/*rangeResiduals=*/N2kInt32NA);
+        // TODO: At present, we can't tel if the satellite is used in the solution.
+        N2kMsg.AddByte(/*tracked=*/0xFF);
+      }
+      for(auto it = sats_glo.begin(); it != sats_glo_end_it; ++it) {
+        N2kMsg.AddByte(/*PRN=*/it->first + 64);
+        // TODO: At present, we can't get azimuth and elevation.
+        // Stefan: not in SBP, but see NMEA 183 message GSV implementation,
+        // we have (should have) some there I think
+        N2kMsg.Add2ByteUDouble(/*elevation=*/N2kDoubleNA, 0.0001, N2kDoubleNA);
+        N2kMsg.Add2ByteUDouble(/*azimuth=*/N2kDoubleNA, 0.0001, N2kDoubleNA);
+        N2kMsg.Add2ByteUDouble(/*CN0=*/it->second * 0.25, 0.01, N2kDoubleNA);
+        // TODO: At present, we can't get range residuals.
+        N2kMsg.Add4ByteUInt(/*rangeResiduals=*/N2kInt32NA);
+        // TODO: At present, we can't tel if the satellite is used in the solution.
+        N2kMsg.AddByte(/*tracked=*/0xFF);
+      }
+      NMEA2000.SendMsg(N2kMsg);
+
+      d << "\tDone.\n";
+    }
+
+    void callback_sbp_heartbeat(u16 sender_id, u8 len, u8 msg[],
+                                void *context) {
+      UNUSED(sender_id);
+      UNUSED(len);
+      UNUSED(msg);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n";
+
+      // This checks for new CAN messages and sends out a heartbeat.
+      NMEA2000.ParseMessages();
+
+      d << "\tDone.\n";
+    }
+}  // namespace
+
+int main(int argc, char *argv[]) {
+  logging_init(cProgramName);
+
+  if (parse_options(argc, argv) != 0) {
+    piksi_log(LOG_ERR, "Invalid arguments.");
+    usage(argv[0]);
+    exit(EXIT_FAILURE);
+  }
+
+  // Get the current state of the CAN interfaces.
+  int state_can0, state_can1;
+  piksi_check(can_get_state(cInterfaceNameCan0, &state_can0),
+              "Could not get %s state.", cInterfaceNameCan0);
+  piksi_check(can_get_state(cInterfaceNameCan1, &state_can1),
+              "Could not get %s state.", cInterfaceNameCan1);
+
+  // Bring the CAN interfaces down if they're up.
+  if(state_can0 != CAN_STATE_STOPPED){
+    piksi_check(can_do_stop(cInterfaceNameCan0),
+                "Could not stop %s.", cInterfaceNameCan0);
+  }
+  if(state_can1 != CAN_STATE_STOPPED){
+    piksi_check(can_do_stop(cInterfaceNameCan1),
+                "Could not stop %s.", cInterfaceNameCan1);
+  }
+
+  // Set the CAN bitrate. This must be done before turning them back on.
+  d << "Setting bitrate to " << bitrate_can0 << " and " << bitrate_can1 << "\n";
+  piksi_check(can_set_bitrate(cInterfaceNameCan0, bitrate_can0),
+              "Could not set bitrate %" PRId32 " on interface %s",
+              bitrate_can0, cInterfaceNameCan0);
+  piksi_check(can_set_bitrate(cInterfaceNameCan1, bitrate_can1),
+              "Could not set bitrate %" PRId32 " on interface %s",
+              bitrate_can1, cInterfaceNameCan1);
+
+  // Set the controlmode to enable flexible datarate (FD) frames.
```

Comment:
If this isn't supported yet, but we still want to check compilation we can add a command line flag for this (but just leave it disabled).